### PR TITLE
Rover: Change the IF statement to a SWITCH statement

### PR DESCRIPTION
**Pull request by muramura**
Clarify the processing for each transition by turning if statements into switch statements.

### CODE REVIEW

File: Rover/mode_auto.cpp [hunk 1]
```diff
@@ -626,16 +626,27 @@ void ModeAuto::exit_mission()
...
 
-    if (g2.mis_done_behave == MIS_DONE_BEHAVE_MANUAL && rover.set_mode(rover.mode_manual, ModeReason::MISSION_END)) {
-        return;
+    case MIS_DONE_BEHAVE_MANUAL:
+        if (rover.set_mode(rover.mode_manual, ModeReason::MISSION_END)) {
+            return;
+        }  
+        break;
+
+    default:
```
Comment (peterbarker) [inline Rover/mode_auto.cpp:648]: Why not handle the other case here and remove `default:`?

**Review by thomasjwilkerson — APPROVED** [review]

**Review by thomasjwilkerson — APPROVED** [review]

File: Rover/mode_auto.cpp (hunk 1 above)
Comment (muramura) [reply Rover/mode_auto.cpp:648]: I agree. I remove.

**Review by khancyr — APPROVED** [review]
LGTM, Peter comment was addressed

**Comment by peterbarker** [comment]
My apologies @muramura , @thomasjwilkerson and @khancyr . 

I'd completely forgotten about this PR.

Yesterday I was separately looking at this code for a client and was annoyed with it enough to patch this myself, and that PR was merged.  I did make this enum-class, too.

Again, apologies for forgetting this PR existed.  Closing this one now.

**Pull request closed without merging**